backend/src/main/java/com/benine/backend/camera/CameraController.java

```diff
@@ -25,11 +25,21 @@
   
   private CameraFactoryProducer camFactoryProducer;
   
+  private Config config;
+  
+  private Logger logger;
+  
+  private StreamController streamController;
+  
   /**
    * Constructor of the camera controller which creates a camera factory producer.
+   * @param serverController this cameraController belongs to.
    */
-  public CameraController() {
-    camFactoryProducer = new CameraFactoryProducer();
+  public CameraController(ServerController serverController) {
+    config = serverController.getConfig();
```

Comment:
A problem with this can occur when the servercontroller decides to use another config or switch to another logger. This change will then not be reflected in these classes. I don't know how big of a problem this will be. But I think it would be good to make these variables (config, logger etc) and the variables in servercontroller final variables. This will ensure they won't (accidentally) diverge. We'll also need to remove the set methods for these final variables.